addons/account/demo/account_demo.py

```diff
@@ -6,187 +6,208 @@
 
 from odoo import api, fields, models, Command
 from odoo.tools.misc import file_open, formatLang
+from odoo.exceptions import UserError, ValidationError
 
 _logger = logging.getLogger(__name__)
 
 
-class AccountChartTemplate(models.Model):
+class AccountChartTemplate(models.AbstractModel):
     _inherit = "account.chart.template"
 
     @api.model
-    def _get_demo_data(self):
+    def _get_demo_data(self, company=False):
         """Generate the demo data related to accounting."""
         # This is a generator because data created here might be referenced by xml_id to data
         # created later but defined in this same function.
-        yield self._get_demo_data_move()
-        yield self._get_demo_data_statement()
-        yield self._get_demo_data_transactions()
-        yield self._get_demo_data_reconcile_model()
-        yield self._get_demo_data_attachment()
-        yield self._get_demo_data_mail_message()
-        yield self._get_demo_data_mail_activity()
+        return {
+            'account.move': self._get_demo_data_move(company),
+            'account.bank.statement': self._get_demo_data_statement(company),
+            'account.bank.statement.line': self._get_demo_data_transactions(company),
+            'account.reconcile.model': self._get_demo_data_reconcile_model(company),
+            'ir.attachment': self._get_demo_data_attachment(company),
+            'mail.message': self._get_demo_data_mail_message(company),
+            'mail.activity': self._get_demo_data_mail_activity(company),
+        }
+
+    def _post_load_demo_data(self, company=False):
+        invoices = (
+            self.ref('demo_invoice_1')
+            + self.ref('demo_invoice_2')
+            + self.ref('demo_invoice_3')
+            + self.ref('demo_invoice_followup')
+            + self.ref('demo_invoice_5')
+            + self.ref('demo_invoice_equipment_purchase')
+        ).with_context(check_move_validity=False)
+
+        # the invoice_extract acts like a placeholder for the OCR to be ran and doesn't contain
+        # any lines yet
+        for move in invoices:
+            try:
+                move.action_post()
+            except (UserError, ValidationError):
+                _logger.exception('Error while posting demo data')
 
     @api.model
-    def _get_demo_data_move(self):
-        cid = self.env.company.id
-        ref = self.env.ref
+    def _get_demo_data_move(self, company=False):
         fifteen_months_ago = fields.Date.today() + relativedelta(months=-15)
-        return ('account.move', {
-            f'{cid}_demo_invoice_1': {
+        return {
+            'demo_invoice_1': {
                 'move_type': 'out_invoice',
-                'partner_id': ref('base.res_partner_12').id,
-                'invoice_user_id': ref('base.user_demo').id,
-                'invoice_payment_term_id': ref('account.account_payment_term_end_following_month').id,
+                'partner_id': 'base.res_partner_12',
```

Comment:
No, one of the only things the loading of the chart of accounts is doing is knowing in which order to load the data.
This allows to inherit all the data easily, and to declare it all at once (i.e. instead of account > tax > account)